honeybadger/contrib/celery.py

```diff
@@ -0,0 +1,94 @@
+from celery import current_task
+from celery.signals import task_failure, task_postrun
+from honeybadger import honeybadger
+from honeybadger.plugins import Plugin, default_plugin_manager
+
+
+class CeleryPlugin(Plugin):
+    def __init__(self):
+        super().__init__("Celery")
+
+    def supports(self, config, context):
+        """
+        Check whether this is a a celery task or not.
+        :param config: honeybadger configuration.
+        :param context: current honeybadger configuration.
+        :return: True if this is a celery task, False else.
+        """
+        return current_task != None
+
+    def generate_payload(self, default_payload, config, context):
+        """
+        Generate payload by checking celery task object.
+        :param context: current context.
+        :param config: honeybadger configuration.
+        :return: a dict with the generated payload.
+        """
+        payload = {
+            "component": current_task.__module__,
+            "action": current_task.name,
+            "params": {
+                "args": list(current_task.request.args),
+                "kwargs": current_task.request.kwargs,
+            },
+            "context": {
+                "task_id": current_task.request.id,
+                "retries": current_task.request.retries,
+                "max_retries": current_task.max_retries,
+            },
+        }
+        default_payload["request"].update(payload)
+        return default_payload
+
+
+class CeleryHoneybadger(object):
+    def __init__(self, app, report_exceptions=False):
+        self.app = app
+        self.report_exceptions = report_exceptions
+        default_plugin_manager.register(CeleryPlugin())
+        if app is not None:
+            self.init_app()
+
+    def init_app(self):
+        """
+        Initialize honeybadger and listen for errors.
+        """
+        self._initialize_honeybadger(self.app.conf)
+        if self.report_exceptions:
+            task_failure.connect(self._on_task_failure, weak=False)
+        task_postrun.connect(self._on_task_postrun, weak=False)
+
+    def _on_task_postrun(self, *args, **kw):
+        """
+        Callback executed after a task is finished.
+        """
+        honeybadger.reset_context()
+
+    def _on_task_failure(self, *args, **kw):
```

Comment:
Can we use the conventional `**kwargs` name for more clarity.

Reply:
Cool, just updated the pr